### ISSUE TITLE
Proposal: `new-line-between-groups` option for `order`

### ISSUE THREAD

**Issue by singles** [benmosher/eslint-plugin-import#282]
This follow up on discussions we had [here](https://github.com/benmosher/eslint-plugin-import/pull/245#issuecomment-211903386) and #246.

Options for that would be either `always` or `never`. 

Then for `always`  and groups like: `groups: [ [ "builtin", "external" ], [ "index", "sibling", "parent" ] ] }` 

So the following would be considered as a problem:

```js
import url from 'url';
import express from 'express'; // <-- missing new line after this statement
import foo from './foo';
```

This would not be considered as a problem:

```js
import url from 'url';
import express from 'express';

import foo from './foo';
```

`new-line-between-groups`  would allow you to configure new lines inside groups like you want - should be pretty flexible. 

I don't think there should be another rule for achieving that, option for `order` IMHO would be fine.

**NOTE**: Option name could of course be changed to something different, `separate-groups(-with-newline)` maybe.

**Comment by benmosher** [comment]
I will mostly defer to @jfmengels and company on this one, but a few gut reactions: 

I think if you don't set the option at all, it shouldn't enforce any spacing.

Also, if you _do_ set it, to either setting, would it enforce that there are no newlines _within_ a group?

i.e.

```js
import url from 'url';
// assuming this were a blank line, would it be reported? "extra newline inside group" or some such
import express from 'express';
```

Also I think it would be optimal if the option were one or two words, something like `newlines`. Clarity over brevity, but brevity would be ideal.

**Comment by jfmengels** [comment]
> I think if you don't set the option at all, it shouldn't enforce any spacing.

:+1:

>  if you do set it, to either setting, would it enforce that there are no newlines within a group?

I think so yes.

Maybe we could specify the number of blank lines between groups?
If I want no blank lines, I can set 0, if I want an empty line, I can set 1, and if I like bigger spacing, I can set 2. If I don't specify it or set it to false (?), nothing will be enforced.

> if the option were one or two words, something like newlines

I think `group-groups` contains the wanted functionality, but it's a bit... repetitive ^^'

**Comment by singles** [comment]
> I think if you don't set the option at all, it shouldn't enforce any spacing.

Yep, then it's do `whatever you want mode` :)

> if you do set it, to either setting, would it enforce that there are no newlines within a group?

Yes, this is what I had in mind.

> Maybe we could specify the number of blank lines between groups?

Personally I don't think it's necessary, especially at the beginning. We might add it later :)

**Comment by jfmengels** [comment]
> > Maybe we could specify the number of blank lines between groups?

> Personally I don't think it's necessary, especially at the beginning. We might add it later :)

One of the things it brings is make the option setting clearer IMO (well, depending on the name of the option...).

**Comment by benmosher** [comment]
I like starting with `true` for 1 line, `false` for 0, and then later if it's int-valued that could be the number of lines, if someone wants that. 

New name idea: `newlines-between`.

**Comment by jfmengels** [comment]
I think I like `newlines-between`. Just one thing, If you only allow at most 1 line and it's a boolean type field, then it should be `newline-between`, which doesn't upgrade well to an eventual int-type later. Man, naming is hard :p

**Comment by singles** [comment]
@jfmengels Depends how to look at that - you can interpret it like: if you will have more than two groups then you will have newline**S** between groups - even if it will be one line between group, total amount of empty lines in imports section will be bigger than 1, thus `newlines`should be fine.

I'm ok with plural form :)

**Comment by singles** [comment]
So, can I assume it's something which would be ok with you guys and I can try to prepare a PR?

**Comment by benmosher** [comment]
I'd wait for an explicit 👍  from @jfmengels but I'm good with where we ended up, discussion-wise.

**Comment by jfmengels** [comment]
Same here :+1:
